apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/ServerExecutor.java

```diff
@@ -0,0 +1,285 @@
+      appRunBuilder.environment().put("BFD_JAVA_HOME", javaHome);
+      serverProcess = appRunBuilder.start();
+
+      appRunConsumer = new ProcessOutputConsumer(serverProcess);
+      Thread appRunConsumerThread = new Thread(appRunConsumer);
+      appRunConsumerThread.start();
+
+      // Await start/finish of application startup by grepping the stdOut for keywords
+      String failureMessage = "Failed startup of context";
+      String successMessage = "Started Jetty.";
+      try {
+        Awaitility.await()
+            .atMost(2, TimeUnit.MINUTES)
+            .until(
+                () ->
+                    appRunConsumer.getStdoutContents().contains(successMessage)
+                        || appRunConsumer.getStdoutContents().contains(failureMessage));
+      } catch (ConditionTimeoutException e) {
+        throw new RuntimeException(
+            "Error: Server failed to start within 120 seconds. STDOUT:\n"
+                + appRunConsumer.getStdoutContents(),
+            e);
+      }
+      // Fail fast if we didn't start the server correctly
+      assertFalse(
+          "Server failed to start due to an error. STDOUT: " + appRunConsumer.getStdoutContents(),
+          appRunConsumer.getStdoutContents().contains(failureMessage));
+      assertTrue(
+          "Did not find the server start message in STDOUT: " + appRunConsumer.getStdoutContents(),
+          appRunConsumer.getStdoutContents().contains(successMessage));
+    }
+    // do nothing if we've already got a server started
+
+    return true;
+  }
+
+  /**
+   * Create a command for the server launcher script to be run with.
+   *
+   * @param workDirectory the server-work directory
+   * @param dbUrl the db url
+   * @param dbUsername the db username
+   * @param dbPassword the db password
+   * @return the command array for the server app
+   */
+  private static String[] createCommandForServerLauncherApp(
+      String workDirectory, String dbUrl, String dbUsername, String dbPassword) {
+    try {
+      Path assemblyDirectory =
+          Files.list(Paths.get(workDirectory))
+              .filter(f -> f.getFileName().toString().startsWith("bfd-server-launcher-"))
+              .findFirst()
+              .orElse(Path.of(""));
+      Path scriptPath = findFirstPathMatchWithFilenameEnd(assemblyDirectory.toString(), ".sh", 1);
+
+      String gcLog = workDirectory + "/gc.log";
+      String maxHeapArg = System.getProperty("its.bfdServer.jvmargs", "-Xmx4g");
+      String containerImageType =
+          System.getProperty("its.testcontainer.db.image", TEST_CONTAINER_DATABASE_IMAGE_DEFAULT);
+      // FUTURE: Inherit these from system properties? Which of these are valuable to pass?
+      String v2Enabled = "true";
+      String pacEnabled = "true";
+      String pacOldMbiHashEnabled = "true";
+      String pacClaimSourceTypes = "fiss,mcs";
+      String includeFakeDrugCode = "true";
+      String includeFakeOrgName = "true";
```

Comment:
Wondering opinions on which of these are useful to parse from the system props. The server script did read these if set, but not sure if it's valuable to be able to adjust these and break the tests which assume a lot of these are the above values in the process.

One example I found while testing is "pacOldMbiHashEnabled" being set to false fails an RDA test which expects it to be true.